tools/backup/backup.sh

```diff
@@ -74,6 +74,19 @@ function cloud_copy() {
     gsutil cp $backup_path $bucket_path
     gsutil cp $backup_path "${bucket_path}${LATEST_POINTER}"
     ;;
+  azure)
+    az storage container create -n "$BUCKET-$database" --account-name $ACCOUNT_NAME --subscription $SUBSCRIPTION
```

Comment:
Not sure about this.   Why do we need to create the storage container here?   If this can be made idempotent so that it doesn't fail if the container already exists, then it could be OK -- leave it and then note in the docs for azure backups that it works this way.

But in general, the data container / bucket / whatever is expected to exist before backup starts.  If it doesn't (on the other clouds) then the backup will fail because the destination can't be written, and it'd be my preference to keep it the same way for Azure.  Require that the user set up the storage destination first.

One of the issues with actually creating the storage container is that it elevates the permissions that are needed for the account doing the backup.  A lot of environments lock this down tightly; and so you might have permissions to upload a file without permissions to create new containers.   Require least privilege possible is the reason to not create this in the backup process.

Reply:
Added a comment in the code. That command can be executed many times and don't fail if it exists. Will give it some more thought if it can be done better.